### ISSUE TITLE
关于复现实验的超参数问题

### ISSUE THREAD

**Issue by FangshengDao** [vla-attack/tex3d#1]
您好，非常感谢您开源这项工作！
我最近在尝试复现论文中的实验，不过发现有些关键的超参数（比如优化迭代次数，学习率，TAAO中用于攻击的帧数等）在论文里没有写得很具体，所以复现起来有点困难。
想请问一下，是否方便提供更完整的超参数配置，比如附录说明或者配置文件？
希望您能为我解惑，万分感激！

**Comment by vla-attack** [comment]
您好，感谢您的关注！以下是我们在主要实验中实际使用的超参数，这些参数基于代码中 `GenerateConfig` 的默认设置：

| Parameter | Argument | Default |
|---|---|---|
| Optimization iterations | `--attack_iters` | `5000` |
| Learning rate | `--attack_lr` | `0.05` (Adam + Cosine Annealing, min `0.005`) |
| Frames used for attack (TAAO) | `--num_frames_to_attack` | `20` |
| TAAO temperature | `--tau` | `1.0` |
| Action loss weight | `--alpha_action` | `1.0` |
| Feature loss weight | `--alpha_feature` | `10.0` |
| Naturalness constraint | `--lambda_nat` | `0.01` |
| Perturbation budget | `--epsilon` | `64/255` (set in `DifferentiableRenderer`) |

我们实验中使用的所有超参数均已预先设定为代码中的默认值，因此大多数情况下您无需进行额外配置即可重现我们的结果。

希望这能有助于你的复制工作！

**Comment by Whc0825** [comment]
请问 --lambda_nat这一超参数在代码中并未看到，它具体是指什么？
我在libero_spatial中使用如下命令去攻击openvla-oft
python experiments/robot/libero/attack_oft.py
--pretrained_checkpoint=/opt/data/private/PythonProjects/tex3d-main/models/openvla-7b-oft-finetuned-libero_spatial
--task_suite_name=libero_spatial
--task_id=0
--object_name=akita_black_bowl
--enable_attack=true
--attack_iters=5000
--attack_lr=0.05
--num_frames_to_attack=20
--num_trials_per_task=50
--local_log_dir=outputs/oft_bowl_spatial_task0
可任务失败率仅有40%左右，并没有达到您论文中提到的78%，这是为什么呢？是因为我哪里设置有问题吗？希望能得到您的指导。

**Comment by FangshengDao** [comment]
> 请问 --lambda_nat这一超参数在代码中并未看到，它具体是指什么？ 我在libero_spatial中使用如下命令去攻击openvla-oft python experiments/robot/libero/attack_oft.py --pretrained_checkpoint=/opt/data/private/PythonProjects/tex3d-main/models/openvla-7b-oft-finetuned-libero_spatial --task_suite_name=libero_spatial --task_id=0 --object_name=akita_black_bowl --enable_attack=true --attack_iters=5000 --attack_lr=0.05 --num_frames_to_attack=20 --num_trials_per_task=50 --local_log_dir=outputs/oft_bowl_spatial_task0 可任务失败率仅有40%左右，并没有达到您论文中提到的78%，这是为什么呢？是因为我哪里设置有问题吗？希望能得到您的指导。

您好，我使用的权重是openvla-7b-oft-libero-spatial，其他超参数和您一致，任务失败率在95%，反而是针对openvla的攻击（spatial任务），任务失败率在20%左右，使用权重为“openvla-7b-finetuned-libero-spatial”。请问您复现的结果中，针对openvla和oft的攻击成功率是否相近？

**Comment by Whc0825** [comment]
您好，我正在尝试复现针对openvla的攻击，目前实验还在运行，我想咨询您一下，这个任务失败率95%指的是全部task的情况下吗？还是仅在某个task上执行的结果。既然您在攻击openvla-oft时使用的权重是没有经过微调的openvla-7b-oft-libero-spatial，为什么在攻击openvla时使用微调以后的openvla-7b-finetuned-libero-spatial权重呢，我也不清楚作者使用的权重具体是哪个。

**Comment by vla-attack** [comment]
您好，我重新在仓库下载了代码，运行的定义参数python experiments/robot/libero/attack_openvla.py
--pretrained_checkpoint=/path_to/openvla-7b-finetuned-libero_spatial
--task_suite_name=libero_spatial
--task_id=0
--object_name=akita_black_bowl
--enable_attack=true
--attack_iters=5000
--attack_lr=0.05
--num_frames_to_attack=20
--num_trials_per_task=50
--local_log_dir=outputs跑了一遍openvla实验，其他的没有做修改，采用的模型是openvla-7b-finetuned-libero-spatial模型，测的是spatial里10个任务的平均结果，结果如下：

<img width="1501" height="576" alt="Image" src="https://github.com/user-attachments/assets/918720ac-c765-484e-8ab1-d20f09a064ca" />

当然，由于环境随机性、模型推理的微小差异等因素，实验结果可能会有一定的波动，但整体来看，结果还是在误差允许范围内的合理结果。

关于你的问题，我感觉可能是环境没有配好，也有可能是没有对10个任务单独分开做对抗攻击训练，也有可能是权重没有下载对，这个是我权重下载的地址：https://huggingface.co/openvla/openvla-7b-finetuned-libero-spatial
建议你排查一下上面这些问题，再次感谢您对我工作的关注，如果还有其他问题，欢迎继续交流！

**Comment by Garsonjw** [comment]
> 您好，我正在尝试复现针对openvla的攻击，目前实验还在运行，我想咨询您一下，这个任务失败率95%指的是全部task的情况下吗？还是仅在某个task上执行的结果。既然您在攻击openvla-oft时使用的权重是没有经过微调的openvla-7b-oft-libero-spatial，为什么在攻击openvla时使用微调以后的openvla-7b-finetuned-libero-spatial权重呢，我也不清楚作者使用的权重具体是哪个。

这里的finetuned或者没有finetuned是官方的命名。我们本质是找在libero上训练过的VLA，这样攻击才有意义。这两个模型都是在libero上训练过，具体模型仓库在issue里已附上~

**Comment by LeoMessiPlayBasketball** [comment]
> 您好，感谢您的关注！以下是我们在主要实验中实际使用的超参数，这些参数基于代码中 `GenerateConfig` 的默认设置：
> 
> Parameter	Argument	Default
> Optimization iterations	`--attack_iters`	`5000`
> Learning rate	`--attack_lr`	`0.05` (Adam + Cosine Annealing, min `0.005`)
> Frames used for attack (TAAO)	`--num_frames_to_attack`	`20`
> TAAO temperature	`--tau`	`1.0`
> Action loss weight	`--alpha_action`	`1.0`
> Feature loss weight	`--alpha_feature`	`10.0`
> Naturalness constraint	`--lambda_nat`	`0.01`
> Perturbation budget	`--epsilon`	`64/255` (set in `DifferentiableRenderer`)
> 我们实验中使用的所有超参数均已预先设定为代码中的默认值，因此大多数情况下您无需进行额外配置即可重现我们的结果。
> 
> 希望这能有助于你的复制工作！

您好，我在复现您论文实验过程中对于oft模型的攻击攻击成功率始终几乎为0%。并且在openvla-finetuned模型中能攻击成功的纹理迁移大oft就没有攻击效果了。我想知道您在评估oft攻击的时候使用的脚本是官方评估脚本吗？是否有状态输入proprio以及是否使用了8个动作块的预测？以及您能不能提供一下训练好的 各物体的攻击纹理以供参考

**Comment by vla-attack** [comment]
> > 您好，感谢您的关注！以下是我们在主要实验中实际使用的超参数，这些参数基于代码中 `GenerateConfig` 的默认设置：
> > Parameter	Argument	Default
> > Optimization iterations	`--attack_iters`	`5000`
> > Learning rate	`--attack_lr`	`0.05` (Adam + Cosine Annealing, min `0.005`)
> > Frames used for attack (TAAO)	`--num_frames_to_attack`	`20`
> > TAAO temperature	`--tau`	`1.0`
> > Action loss weight	`--alpha_action`	`1.0`
> > Feature loss weight	`--alpha_feature`	`10.0`
> > Naturalness constraint	`--lambda_nat`	`0.01`
> > Perturbation budget	`--epsilon`	`64/255` (set in `DifferentiableRenderer`)
> > 我们实验中使用的所有超参数均已预先设定为代码中的默认值，因此大多数情况下您无需进行额外配置即可重现我们的结果。
> > 希望这能有助于你的复制工作！
> 
> 您好，我在复现您论文实验过程中对于oft模型的攻击攻击成功率始终几乎为0%。并且在openvla-finetuned模型中能攻击成功的纹理迁移大oft就没有攻击效果了。我想知道您在评估oft攻击的时候使用的脚本是官方评估脚本吗？是否有状态输入proprio以及是否使用了8个动作块的预测？以及您能不能提供一下训练好的 各物体的攻击纹理以供参考

您好，感谢关注。
仓库中已经提供了论文实验所使用的评估代码，具体的评估流程、状态输入以及动作预测设置，可以参考仓库中的实现。该代码并不是直接沿用原始模型的官方评估脚本，而是针对我们的对抗纹理训练方式进行了相应调整。
具体来说，我们在评估阶段沿用了训练阶段的双渲染器对齐方式，使模型在测试时接收到的输入与对抗训练阶段保持一致。这样能够尽量避免训练与测试之间的输入分布偏移，并更准确地反映对抗纹理的攻击效果。如果直接使用其他评估脚本，可能会导致模型测试时的输入分布与对抗训练阶段不一致，从而使攻击成功率明显下降。
因此，建议您按照仓库中提供的评估代码和配置进行复现。关于是否包含 proprio 状态输入以及是否采用8个动作块预测，也请以仓库代码中的具体实现为准，这与我们论文实验中的设置一致。
此外，我们后续会将训练好的各物体对抗纹理上传至仓库，供大家复现和参考。
